### PR TITLE
Updating links to docs.rs or gh-pages

### PR DESCRIPTION
**Pull request by robertbastian**
Links on internal docs point to the gh-pages website (so they always point to main's latest docs), while links on external docs point to docs.rs

### CODE REVIEW

**Review by Manishearth — APPROVED** [review]
Kinda feel like FFI should use up to date docs but not a strong opinion

**Comment by robertbastian** [comment]
Why? It's versioned with our crates.io releases, so the docs should be as well.

**Comment by Manishearth** [comment]
Hm, fair.

**Review by Manishearth — DISMISSED** [review]

File: docs/design/properties_code_point_trie.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ In the binary properties case, we can reduce the scope to there being only two d
 
 The use cases, or manner of data access, inform the designs of APIs and data structures.
 For regular expression parsers (regex), we need to support a text description of a _set of code points_ sharing a property.
-In this case, returning a [`CodePointInversionList`](https://icu4x.unicode.org/doc/icu/collections/codepointinvlist/struct.CodePointInversionList.html) (a set of code points, a.k.a. [`UnicodeSet`](https://unicode-org.github.io/icu/userguide/strings/unicodeset.html) in ICU) provides the most efficient usable data.
```
Comment (sffc) [inline docs/design/properties_code_point_trie.md:45]: Issue: I would really like the canonical docs link to be on `icu4x.unicode.org` (or docs.rs).

**Review by sffc — APPROVED** [review]
This is a net improvement but I would still like to see icu4x.unicode.org be the canonical URL

File: docs/design/properties_code_point_trie.md (hunk 1 above)
Comment (robertbastian) [reply docs/design/properties_code_point_trie.md:45]: Why docs.rs? Internal documentation might point at things that are not published yet.

I'm not sure I agree that `icu4x.unicode.org` would be better. The `github.io` domain kind of signals that these are dev artifacts and it's not a polished release.

Comment (sffc) [reply docs/design/properties_code_point_trie.md:45]: This PR correctly contains some links to docs.rs.

Fair point about `github.io`. We don't need to make that call right now.

**Review by Manishearth — APPROVED** [review]

**Pull request merged**